HomeWork by Serhii Kohut

Serhii-Kohut opened the pull request:
Added my code of ArticleAnalyzer

Serhii-Kohut committed:
init commit

Serhii-Kohut committed:
changed logic and structure of project

Serhii-Kohut committed:
commit with changes

Serhii-Kohut committed:
changed dependencies

Serhii-Kohut committed:
changed CATEGORIES_FILE_PATH for categories.json

Serhii-Kohut committed:
update equals getSenderIdentifier with getNotificationType (enum)

Serhii-Kohut committed:
updated method process in ContentProcessorServiceImpl

Serhii-Kohut commented:
added 4 new commits: fixed: 
- creating categories.json in resouces;
- equals getSenderIdentifier with getNotificationType (enum)
- updated method process in ContentProcessorServiceImpl.

vitaliiStefanchakInventorsoft commented on article-analyzer/build.gradle:
```diff
@@ -0,0 +1,52 @@
+
+dependencies {
+    compileOnly 'org.projectlombok:lombok:1.18.30'
+  //  implementation 'org.springframework.boot:spring-boot-starter-web'
```
commented code - bad code, remove pls)

Serhii-Kohut replied:
done

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/co/inventorsoft/academy/homework/ArticleAnalyzerApplication.java:
```diff
@@ -0,0 +1,54 @@
+    private final UserReader userReader;
+    private final List<NotificationService> notificationServiceList;
+
+    @EventListener(ContextRefreshedEvent.class)
+    public void runOnStartUp() {
+        List<Article> articleList = articleReader.jsonToListArticles(ARTICLES_FILE);
+        Set<String> categories = contentProcessorServiceImpl.process(articleList);
+        categorySaverServiceImpl.saveCategory(categories);
+
+        List<User> userList = userReader.jsonToListUsers(USERS_FILE);
+
+        for (User user : userList) {
+            for (NotificationService notificationService : notificationServiceList) {
+                if (notificationService.getSenderIdentifier().equals(user.getNotificationType().name())) {
+                    for (Article article : articleList) {
+                        notificationService.notify(user, article);
+                    }
+                }
+            }
+        }
+    }
```
this is a pretty interesting idea, but do not write any business logic inside the infrastructure code, The main class and method are considered as such, your main method should just run one void method like `analyzeArticles`, hence do create some ArticleAnalyzerService or whatever you name it and move the whole code from main class into there

Serhii-Kohut replied:
done

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/co/inventorsoft/academy/homework/nofifier/impl/SlackNotificationServiceImpl.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public String getSenderIdentifier() {
+        return "SLACK";
```
you may return NotificationType.SLACK here, then it will be easier to compare

vitaliiStefanchakInventorsoft replied:
same to email

Serhii-Kohut replied:
done

vitaliiStefanchakInventorsoft commented on article-analyzer/src/main/java/co/inventorsoft/academy/homework/service/impl/CategorySaverServiceImpl.java:
```diff
@@ -0,0 +1,62 @@
+@Component
+@AllArgsConstructor
+public class CategorySaverServiceImpl implements CategorySaverService {
+    private static final String CATEGORIES_FILE_PATH = "article-analyzer/src/main/resources/categories.json";
```
it is a good fit for moving the constant into application.properties, the same for articles and users

Serhii-Kohut replied:
done

vitaliiStefanchakInventorsoft commented on src/main/resources/categories.json:
```diff
@@ -0,0 +1 @@
+["you"]
```
remove this folder and file, it should be under article-analyzer/src/main/resoures

Serhii-Kohut replied:
done

Serhii-Kohut committed:
build.gradle - removed unused dependencies

removed wrong
categories.json

Serhii-Kohut committed:
ArticleAnalyzerApplication: removed business logic

added AnalyzeArticle.java with business logic from Main class

Serhii-Kohut committed:
NotificationService - changed return type and methods

Serhii-Kohut committed:
GsonConfig renamed to SpringConfig
SpringConfig: added categoriesFilePath

moved the constant into application.properties and changed fields

Serhii-Kohut commented:
fixed all comments and problems

vitaliiStefanchakInventorsoft commented:
![ConsiderItDoneItsDoneGIF (2)](https://github.com/vitaliiStefanchakInventorsoft/spring-lecture/assets/103140969/0a07a22d-a155-4c86-981b-77f0c53c5f4c)
